drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -13,6 +13,26 @@
 LOG_MODULE_REGISTER(lis2dh, CONFIG_SENSOR_LOG_LEVEL);
 #include "lis2dh.h"
 
+#if defined(DT_INST_0_ST_LSM303AGR_ACCEL)
+/*
+ * Use values for 16g low-power mode in LSM303AGR DS "Table 3. Sensor
+ * characteristics" and LIS2DH DS "Table 3. Mechanical characteristics".
+ */
+#define ACCEL_SCALE_FACTOR(x)		((x) * 18758LL / 19200LL)
```

Comment:
So you decided to use a matematic expression.
Actually I preferred to have a separate set of scales for the two different sensors.
I think it is easier to find the correspondent values on the datasheets.
What do you think?

Reply:
```
#define ACCEL_SCALE(sensitivity)		\
	(SENSOR_G * (sensitivity) >> 14 / 100)

/*
 * Use values for low-power mode in DS "Mechanical (Sensor) characteristics",
 * multiplied by 100.
 */
static const u32_t lis2dh_reg_val_to_scale[] = {
#if defined(DT_INST_0_ST_LSM303AGR_ACCEL)
	ACCEL_SCALE(1563),
	ACCEL_SCALE(3126),
	ACCEL_SCALE(6252),
	ACCEL_SCALE(18758),
#else
	ACCEL_SCALE(1600),
	ACCEL_SCALE(3200),
	ACCEL_SCALE(6400),
	ACCEL_SCALE(19200),
#endif
};
```

Something like that? :)

Reply:
> ```
> #define ACCEL_SCALE(sensitivity)		\
> 	(SENSOR_G * (sensitivity) >> 14 / 100)
> 
> /*
>  * Use values for low-power mode in DS "Mechanical (Sensor) characteristics",
>  * multiplied by 100.
>  */
> static const u32_t lis2dh_reg_val_to_scale[] = {
> #if defined(DT_INST_0_ST_LSM303AGR_ACCEL)
> 	ACCEL_SCALE(1563),
> 	ACCEL_SCALE(3126),
> 	ACCEL_SCALE(6252),
> 	ACCEL_SCALE(18758),
> #else
> 	ACCEL_SCALE(1600),
> 	ACCEL_SCALE(3200),
> 	ACCEL_SCALE(6400),
> 	ACCEL_SCALE(19200),
> #endif
> };
> ```
> 
> 
> Something like that? :)

Yes, exactly! :)

Reply:
@mniestroj 
So, are you changing it in this way?

Reply:
@avisconti Yes, just wanted to test with my board before submission :) It is ready now.